modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java

```diff
@@ -405,6 +406,52 @@ public void testPutObjectDefinition() throws Exception {
 		_objectDefinitionLocalService.deleteObjectDefinition(
 			postObjectDefinition.getId());
 
+		// Account entry update with null as objectDefinitionId
+
+		com.liferay.object.model.ObjectDefinition
+			serviceBuilderAccountEntryObjectDefinition2 =
+				_objectDefinitionLocalService.fetchSystemObjectDefinition(
+					AccountEntry.class.getSimpleName());
+
+		ObjectDefinition accountSystemObjectDefinition =
+			objectDefinitionResource.getObjectDefinition(
+				serviceBuilderAccountEntryObjectDefinition2.
+					getObjectDefinitionId());
+
+		String titleObjectFieldName = "type";
+		String externalReferenceCode = RandomTestUtil.randomString();
+
+		accountSystemObjectDefinition.setTitleObjectFieldName(
+			titleObjectFieldName);
+		accountSystemObjectDefinition.setExternalReferenceCode(
+			externalReferenceCode);
+
+		ObjectRelationship objectRelationship =
+			accountSystemObjectDefinition.getObjectRelationships()[0];
+
+		objectRelationship.setObjectDefinitionId2((Long)null);
+
+		accountSystemObjectDefinition.setObjectRelationships(
+			new ObjectRelationship[]{objectRelationship});
+
+		ObjectDefinition accountEntryObjectDefinitionResult =
+			objectDefinitionResource.putObjectDefinition(
+				accountSystemObjectDefinition.getId(),
+				accountSystemObjectDefinition);
+
+		Assert.assertEquals(
+			accountEntryObjectDefinitionResult.getTitleObjectFieldName(),
+			titleObjectFieldName);
+
+		Assert.assertEquals(
+			accountEntryObjectDefinitionResult.getExternalReferenceCode(),
+			accountSystemObjectDefinition.getExternalReferenceCode());
+
+		ObjectRelationship objectRelationshipResult =
+			accountEntryObjectDefinitionResult.getObjectRelationships()[0];
+
+		Assert.assertNotNull(objectRelationshipResult.getObjectDefinitionId2());
+
```

Comment:
@evibudai please simply the code to:

```
// Account entry update with null objectDefinitionId2

accountEntryObjectDefinition.setExternalReferenceCode(
	RandomTestUtil.randomString());
accountEntryObjectDefinition.setTitleObjectFieldName(
	"type");

ObjectRelationship objectRelationship =
	accountEntryObjectDefinition.getObjectRelationships()[0];

objectRelationship.setObjectDefinitionId2((Long)null);

ObjectDefinition putAccountEntryObjectDefinition =
	objectDefinitionResource.putObjectDefinition(
		accountEntryObjectDefinition.getId(),
		accountEntryObjectDefinition);

Assert.assertEquals(
	accountEntryObjectDefinition.getExternalReferenceCode(),
	putAccountEntryObjectDefinition.getExternalReferenceCode());
Assert.assertEquals(
	accountEntryObjectDefinition.getTitleObjectFieldName(),
	putAccountEntryObjectDefinition.getTitleObjectFieldName());

objectRelationship =
	putAccountEntryObjectDefinition.getObjectRelationships()[0];

Assert.assertNotNull(objectRelationship.getObjectDefinitionId2());

```

My changes:
- I reused the existing variable `accountEntryObjectDefinition`. 
- I ordered alphabetically: `externalReferenceCode` before `titleObjectFieldName`. 
- In the `Assert.assertEquals`, we must place the expected result before the actual result.

Futhermore, after we do:

```
ObjectRelationship objectRelationship =
	accountEntryObjectDefinition.getObjectRelationships()[0];

objectRelationship.setObjectDefinitionId2((Long)null);
```
we don't need to do:

```
accountEntryObjectDefinition.setObjectRelationships(
	new ObjectRelationship[]{objectRelationship});
```

the `objectRelationship` is already the same object inside the `accountEntryObjectDefinition` object.

Thanks!